app/classes/pattern_search/observation.rb

```diff
@@ -66,16 +66,43 @@ def model
 
     def build_query
       super
+      hack_name_query
+      default_to_including_synonyms_and_subtaxa
+    end
+
+    private
+
+    # Temporary hack to get include_subtaxa/synonyms to work.
+    # Will rip out when we do away with pattern search query flavor.
+    # This converts any search that *looks like* a name search into
+    # an actual name search.
+    def hack_name_query
+      return unless flavor == :pattern_search &&
+                    args[:names].empty? &&
+                    (is_pattern_a_name? || any_taxa_modifiers_present?)
+
+      self.flavor = :all
+      args[:names] = args[:pattern]
+      args.delete(:pattern)
+    end
+
+    def default_to_including_synonyms_and_subtaxa
+      return if args[:names].empty?
+
+      args[:include_subtaxa] = true if args[:include_subtaxa].nil?
+      args[:include_synonyms] = true if args[:include_synonyms].nil?
+    end
+
+    def is_pattern_a_name?
+      ::Name.where("text_name = ? OR search_name = ?",
+                   args[:pattern].to_s, args[:pattern].to_s).any?
+    end
 
-      # Temporary hack to get include_subtaxa/synonyms to work.
-      # Will rip out when we do away with pattern search query flavor.
-      if flavor == :pattern_search &&
-         (!args[:include_subtaxa].nil? || !args[:include_synonyms].nil? ||
-          !args[:include_all_name_proposals].nil?)
-        self.flavor = :all
-        args[:names] = args[:pattern]
-        args.delete(:pattern)
-      end
+    def any_taxa_modifiers_present?
+      !args[:include_subtaxa].nil? ||
+        !args[:include_synonyms].nil? ||
+        !args[:include_all_name_proposals].nil? ||
+        !args[:exclude_consensus].nil?
```

Comment:
Could any of these be empty, but not nil? 
If so, then 
```suggestion
      args[:include_subtaxa].present? ||
      args[:include_synonyms].present? ||
      args[:include_all_name_proposals].present?
      args[:exclude_consensus].present?
```
If not, will `args[x]` work in place of `!args[x].nil?`?
```suggestion
      args[:include_subtaxa] ||
      args[:include_synonyms] ||
      args[:include_all_name_proposals] ||
      args[:exclude_consensus]
```

Reply:
As I understand it args[:boolean] can have three states: nil, true or false.  Therefore neither of your proposals would work, I'm afraid.  Actually, I already had tried your first proposal!  (great minds obviously think alike)  I suppose it *could* be re-written as !(a.nil? && b.nil? && c.nil?).  But that doesn't really feel any better to me.

Reply:
Thanks for the explanation. I hadn't realized what you were trying to accomplish.